Add autocorrect for Style/Lambda cop

lumeet opened the pull request:
Fixes #1272. I ended up using sub instead of `Parser::Source::Rewriter` just because I feel it's still more readable that way. I'm not quite sure if there's a strong preference for the other option, though. For sure, there's also the possibility that I'm missing something in the pattern. :)

jonas054 commented on lib/rubocop/cop/style/lambda.rb:
```diff
@@ -38,6 +38,19 @@ def lambda_length(block_node)
+
+          @corrections << lambda do |corrector|
+            src = if node.loc.expression.source == 'lambda'
+                    expr.source.sub(/^lambda\s+\{\s*\|([^\|]+)\|/, '->(\\1) {')
```
Lambdas taking no arguments are not supported here, so they will be left unchanged. But the report will still say `[Corrected]`. I think the best solution would be to have a special message for the no arguments case: `Use the new lambda literal syntax -> {...}.`, and then to handle it in `autocorrect` as well.

bbatsov replied:
Yeah, you're right.

lumeet replied:
Good point. Suddenly I'm not so sure about the regex strategy anymore. :) Do I understand correctly that you're recommending me to add the new message within the scope of this issue? Sure, if that's the case.

jonas054 replied:
I was thinking that since you have to introduce special handling of the zero args case in `autocorrect` you might as well change the message too.

lumeet replied:
Absolutely. I was just unsure.

jonas054 commented on CHANGELOG.md:
```diff
@@ -7,6 +7,7 @@
 * [#1430](https://github.com/bbatsov/rubocop/issues/1430): Add `--except` option for disabling cops on the command line. ([@jonas054][])
 * [#1506](https://github.com/bbatsov/rubocop/pull/1506): Add auto-correct from `EvenOdd` cop. ([@blainesch][])
 * [#1507](https://github.com/bbatsov/rubocop/issues/1507): `Debugger` cop now checks for the Capybara debug methods `save_and_open_page` and `save_and_open_screenshot`. ([@rrosenblum][])
+* [#1271](https://github.com/bbatsov/rubocop/issues/1272): `Lambda` cop does auto-correction. ([@lumeet][])
```
Typo in the URL, should be `1271`. And you need to add a link for yourself at the bottom of the file.

lumeet replied:
Oops, will be fixed.

lumeet commented:
Oh no, now I'm fumbling with rebase...

lumeet commented:
Now it should be alright. Sorry about the mess, @jonas054 and @bbatsov. I replaced the regexps with something more elegent (I hope), added some tests and added the new message.

bbatsov commented:
The commits have to squashed together. Apart from this, the new version looks good to me.

bbatsov commented:
It also seems you'll have to rebase the PR on top of the current `master`.

bbatsov commented:
:+1: Thanks!
